File: src/main/java/xyz/upperlevel/hgame/networking/NettyChannelEventCaller.java
package xyz.upperlevel.hgame.networking;

import io.netty.channel.*;

public class NettyChannelEventCaller extends SimpleChannelInboundHandler<PayloadPacket> {
    private final Endpoint endpoint;

    public NettyChannelEventCaller(Endpoint endpoint) {
        this.endpoint = endpoint;
    }

    @Override
    public void channelActive(final ChannelHandlerContext ctx) {
        endpoint.onOpen(ctx.channel());
        /*if (!onOpen()) {
            ctx.close();
        }*/
    }

    @Override
    public void channelInactive(final ChannelHandlerContext ctx) {
        endpoint.onClose();
    }

    @Override
    protected void channelRead0(ChannelHandlerContext ctx, PayloadPacket msg) throws Exception {
        endpoint.onPacketReceive(msg);
    }
}

File: src/main/java/xyz/upperlevel/hgame/network/NettyChannelEventCaller.kt
package xyz.upperlevel.hgame.network

import io.netty.channel.ChannelHandlerContext
import io.netty.channel.SimpleChannelInboundHandler

class NettyChannelEventCaller(private val endpoint: Endpoint) : SimpleChannelInboundHandler<PayloadPacket>() {

    override fun channelActive(ctx: ChannelHandlerContext) {
        endpoint.onOpen(ctx.channel())
        /*if (!onOpen()) {
            ctx.close();
        }*/
    }

    override fun channelInactive(ctx: ChannelHandlerContext) {
        endpoint.onClose()
    }

    override fun channelRead0(ctx: ChannelHandlerContext, msg: PayloadPacket) {
        endpoint.onPacketReceive(msg)
    }
}
